shared/lib_battery.cpp

```diff
@@ -530,72 +557,116 @@ void voltage_table_t::copy(voltage_t * voltage)
 	*/
 }
 
+double voltage_table_t::calculate_voltage(double DOD) {
+    DOD = fmax(0., DOD);
+    DOD = fmin(DOD, 100.);
 
-void voltage_table_t::updateVoltage(capacity_t * capacity, thermal_t * , double )
-{
-	double cell_voltage = _cell_voltage;
-	double DOD = capacity->DOD();
-	double I_string = capacity->I() / _num_strings;
-	double DOD_lo, DOD_hi, V_lo, V_hi;
-	bool voltage_found = exactVoltageFound(DOD, cell_voltage);
-	if (!voltage_found)
-	{
-		prepareInterpolation(DOD_lo, V_lo, DOD_hi, V_hi, DOD);
-		cell_voltage = util::interpolate(DOD_lo, V_lo, DOD_hi, V_hi, DOD) - I_string * _R;
-	}
+    size_t row = 0;
+    while (row < m_voltage_table.size() && DOD > m_voltage_table[row].first){
+        row++;
+    }
 
-	// the cell voltage should not increase when the battery is discharging
-	if (I_string <= 0 || (I_string > 0 && cell_voltage <= _cell_voltage))
-		_cell_voltage = cell_voltage;
-	
+    return fmax(slopes[row] * DOD + intercepts[row], 0);
 }
 
-bool voltage_table_t::exactVoltageFound(double DOD, double & V)
-{
-	bool contained = false;
-	for (size_t r = 0; r != _voltage_table.size(); r++)
-	{
-		if (_voltage_table[r].DOD() == DOD)
-		{
-			V = _voltage_table[r].V();
-			contained = true;
-			break;
-		}
-	}
-	return contained;
+double voltage_table_t::calculate_voltage_for_current(double I, double q, double qmax, double) {
```

Comment:
the virtual function has (I, q, qmax and temp) as parameters since the vanadium flow model requires temp but the other voltage models do not. so when it's not required, the name of the double is removed to show that it's not being used (otherwise there's a 'unused parameter' warning)